perf(oxfmt): Use `worker_threads` by `tinypool` for prettier formatting

leaysgur opened the pull request:
Part of #16606 

- Use `worker_threads` via `tinypool` lib
  - Until now, the Rust side had been operating entirely in a multithreaded manner, while the JS side remained single-threaded, resulting in sequential execution and waiting
  - Number of threads are the same value for Rust side
- Also can be applied to embedded formatting path
- Since `tinypool` couldn't be bundled, defined as `dependencies` in the `npm/oxfmt/package.json`

Dunqing reviewed (changes requested):
Overall, it looks good! Just one thing that I do not know if it will work. Feel free to do it in the follow-up PR if you want.

leaysgur commented on apps/oxfmt/src-js/prettier-worker.ts:
```diff
@@ -0,0 +1,62 @@
+// Initialize config from `workerData` (passed during pool creation)
+const [, { configJSON }] = workerData satisfies [unknown, WorkerData];
+// SAFETY: Always valid JSON constructed by Rust side
+const configCache = JSON.parse(configJSON) as Options;
```
Thanks! let me check the reference~ 👌🏻

leaysgur replied:
Yup, You are correct. Updated.

Dunqing reviewed (approved):
Great!

leaysgur committed:
perf(oxfmt): Use `worker_threads` by `tinypool` for prettier formatting (#16618)

Part of #16606

- Use `worker_threads` via `tinypool` lib
  - Until now, the Rust side had been operating entirely in a multithreaded manner, while the JS side remained single-threaded, resulting in sequential execution and waiting
  - Number of threads are the same value for Rust side
- Also can be applied to embedded formatting path
- Since `tinypool` couldn't be bundled, defined as `dependencies` in the `npm/oxfmt/package.json`
